GVFS/GVFS.Platform.Mac/MacFileSystemVirtualizer.cs

```diff
@@ -218,7 +218,7 @@ protected override bool TryStart(out string error)
             this.virtualizationInstance.OnPreDelete = this.OnPreDelete;
             this.virtualizationInstance.OnNewFileCreated = this.OnNewFileCreated;
             this.virtualizationInstance.OnFileRenamed = this.OnFileRenamed;
-            this.virtualizationInstance.OnHardLinkCreated = this.OnHardLinkCreated;
+            this.virtualizationInstance.OnHardLinkCreated = this.NotifyHardLinkCreated;
```

Comment:
Otherwise we clash with the base class's OnHardLinkCreated.  
This seems to be the way NotifyFilePreConvertToFull got around the same problem.

Reply:
..but since `NotifyHardLinkCreated` just calls through to `this.OnHardLinkCreated` why does `NotifyHardLinkCreated` need to exist anyway? Couldn't we just do..

```suggestion
            this.virtualizationInstance.OnHardLinkCreated = this.OnHardLinkCreated;
```

Reply:
> ..but since NotifyHardLinkCreated just calls through to this.OnHardLinkCreated why does NotifyHardLinkCreated need to exist anyway? Couldn't we just do..

That's a good point.  I think the parameters here would need to be re-ordered to match `OnHardLinkCreated`:

`NotifyHardLinkCreated(string relativeNewLinkPath, string relativeFromPath)` - > `NotifyHardLinkCreated(string relativeFromPath, string relativeNewLinkPath)`

But that seems like a good change anyway as it would make the API more consistent with Windows:

```
        private void NotifyHardlinkCreated(
            string relativeExistingFilePath,
            string relativeNewLinkPath,
            uint triggeringProcessId,
            string triggeringProcessImageFileName)
        {
            this.OnHardLinkCreated(relativeExistingFilePath, relativeNewLinkPath);
        }
```

> This seems to be the way NotifyFilePreConvertToFull got around the same problem.

If we tweaked the Mac's managed API for `NotifyFilePreConvertToFull` to return `true`/`false` we could use @mjcheetham's suggestion there too, but I'd also be OK leaving that as-is as the code is a bit unrelated to this PR.

Reply:
I think in this case less code is good so I've updated with @mjcheetham 's suggestion to call `this.OnHardLinkCreated` .  Thank you @wilbaker for calling out that the params are switched, i'm sure i would have fallen for that!